src/audio/kpb.c

```diff
@@ -366,57 +368,62 @@ static int kpb_copy(struct comp_dev *dev)
 	struct comp_data *kpb = comp_get_drvdata(dev);
 	struct comp_buffer *source;
 	struct comp_buffer *sink;
+	size_t copy_bytes = 0;
+	static size_t buffered;
```

Comment:
@tlauda, I just increment buffered until it is equal to maximum size of history buffer. Once it is equal we set the flag saying buffer is full and then no longer increment it - no need for that. Although I think it would be better to move buffered to comp_data and reset in kpb_reset().